### PR TITLE
Fast path state machine

### PR DESCRIPTION
**Pull request by benaadams**

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs [hunk 1]
```diff
@@ -43,10 +43,14 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider
             // All done if request services is set
             if (existingFeature?.RequestServices != null)
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs]: 99% of the time this code path will never get hit.

Comment (benaadams) [inline src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs]: Depends if is skippable for benchmark I suppose? Stepping through profile traces line by line atm

Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs]: You can just remove the whole thing in the bemchmark.

Comment (benaadams) [inline src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerMiddleware.cs]: That's probably an even faster path :+1:

**Pull request closed without merging**